xarray/tests/test_dataset.py

```diff
@@ -1137,6 +1137,94 @@ def test_chunks_does_not_load_data(self) -> None:
 
     @requires_dask
     def test_chunk(self) -> None:
+        data = create_test_data()
+        for chunks in [1, 2, 3, 4, 5]:
+            rechunked = data.chunk({"dim1": chunks})
+            assert rechunked.chunks["dim1"] == (chunks,) * (8 // chunks) + (
+                (8 % chunks,) if 8 % chunks else ()
+            )
+
+            rechunked = data.chunk({"dim2": chunks})
+            assert rechunked.chunks["dim2"] == (chunks,) * (9 // chunks) + (
+                (9 % chunks,) if 9 % chunks else ()
+            )
+
+            rechunked = data.chunk({"dim1": chunks, "dim2": chunks})
+            assert rechunked.chunks["dim1"] == (chunks,) * (8 // chunks) + (
+                (8 % chunks,) if 8 % chunks else ()
+            )
+            assert rechunked.chunks["dim2"] == (chunks,) * (9 // chunks) + (
+                (9 % chunks,) if 9 % chunks else ()
+            )
+
+    @requires_dask
+    @pytest.mark.parametrize("use_cftime", [True, False])
+    def test_chunk_by_season_resampler(self, use_cftime: bool) -> None:
+        """Test chunking using SeasonResampler."""
+        import dask.array
+
+        if use_cftime:
+            pytest.importorskip("cftime")
+
+        N = 365 * 2  # 2 years
+        if use_cftime:
+            time = xr.date_range("2001-01-01", periods=N, freq="D", use_cftime=True)
+        else:
+            time = xr.date_range("2001-01-01", periods=N, freq="D")
```

Comment:
```suggestion
        N = 366 + 365  # 2 years
        time = xr.date_range("2000-01-01", periods=N, freq="D", use_cftime=use_cftime)
```

By starting in 2000, we can check leap year logic

Reply:
Can we parameterize this over calendars too? `360_day`, `noleap` and `standard` should be good enough.

Reply:
Changed to `N = 366 + 365` starting from 2000-01-01 for leap year testing. Added parameterization over calendars: standard, noleap, 360_day.